src/api/graph/graph.controller.spec.ts

```diff
@@ -0,0 +1,100 @@
+import { CacheModule } from '@nestjs/cache-manager';
+import { INestApplication } from '@nestjs/common';
+import { Test, TestingModule } from '@nestjs/testing';
+import * as request from 'supertest';
+
+import { GraphController } from '@/api/graph/graph.controller';
+import { GraphType, TimeFrame } from '@/api/graph/graph.model';
+import { GraphService } from '@/api/graph/graph.service';
+import { overviewMaxTVLGraph } from '@/test/mock-data/graph-mock-data';
+
+const mockGraphService = {
+  getGraph: jest.fn(),
+};
+
+describe('GraphController', () => {
+  let app: INestApplication;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      controllers: [GraphController],
+      providers: [
+        {
+          provide: GraphService,
+          useValue: mockGraphService,
+        },
+      ],
+      imports: [
+        CacheModule.register({
+          isGlobal: true,
+        }),
+      ],
+    }).compile();
+
+    app = module.createNestApplication();
+    await app.init();
+  });
+
+  describe('GET /graph', () => {
```

Comment:
maybe we could leave a it.todo or something with more in-depth tests

Reply:
✅